app/code/Magento/Catalog/Pricing/Render/FinalPriceBox.php

```diff
@@ -64,26 +64,36 @@ public function __construct(
      */
     protected function _toHtml()
     {
-        if (!$this->salableResolver->isSalable($this->getSaleableItem())) {
-            return '';
-        }
-
         $result = parent::_toHtml();
-        //Renders MSRP in case it is enabled
-        if ($this->isMsrpPriceApplicable()) {
-            /** @var BasePriceBox $msrpBlock */
-            $msrpBlock = $this->rendererPool->createPriceRender(
-                MsrpPrice::PRICE_CODE,
-                $this->getSaleableItem(),
-                [
-                    'real_price_html' => $result,
-                    'zone' => $this->getZone(),
-                ]
-            );
-            $result = $msrpBlock->toHtml();
-        }
 
-        return $this->wrapResult($result);
+        if(!$result) {
+            $result = BasePriceBox::_toHtml();
```

Comment:
`Magento\Catalog\Pricing\Render\FinalPriceBox` extends from `Magento\Framework\Pricing\Render\PriceBox` (`BasePriceBox`).

If I'm not mistaken, with the above in the mind the parent class is actually called twice, once through `parent::_toHtml` and once through `BasePriceBox::_toHtml`, is there a reason for this and thus on purpose or not?